src/plasma/malloc.cc

```diff
@@ -111,12 +108,9 @@ void *fake_mmap(size_t size) {
   /* Increase dlmalloc's allocation granularity directly. */
   mparams.granularity *= GRANULARITY_MULTIPLIER;
 
-  struct mmap_record *record = malloc(sizeof(struct mmap_record));
-  record->fd = fd;
-  record->pointer = pointer;
-  record->size = size;
-  HASH_ADD(hh_fd, records_by_fd, fd, sizeof(fd), record);
-  HASH_ADD(hh_pointer, records_by_pointer, pointer, sizeof(pointer), record);
+  mmap_record &record = mmap_records[pointer];
```

Comment:
Yeah, this creates a new mmap_record in the hashtable and then sets a reference to that entry, the mmap_record that was created in the hashtable is then modified through that reference.